client/src/modules/dashboard/index.jsx

```diff
@@ -59,9 +65,27 @@ class Dashboard extends Component {
 
   hideDialog = () => this.handleDialogContext(null)();
 
+  handleArchivedListsVisibility = () => {
+    this.setState(({ areArchivedListVisible }) => ({
+      areArchivedListVisible: !areArchivedListVisible
+    }));
+    this.handleArchivedListsData();
+  };
+
+  handleArchivedListsData = id => {
+    const { areArchivedListVisible } = this.state;
+    const {
+      fetchArchivedListsMetaData,
+      removeArchivedListsMetaData
+    } = this.props;
+    !areArchivedListVisible
```

Comment:
as Marek said, no teranry when nothing is return 
do something like. And try start with positive condition first not negative `!`
```
const action = areAchivedListVisible ? removeArchivedListsMetaData : fetchArchivedListsMetaData
action();
```